core/globals.h

```diff
@@ -912,7 +912,15 @@ struct _dcontext_t {
 
     dr_where_am_i_t whereami; /* where control is at the moment */
 #ifdef UNIX
+/* On ARM based machines, char is unsigned by default.
+ * https://www.arm.linux.org.uk/docs/faqs/signedchar.php
+ * But we need signed char, so we explicitly qualify this declaration.
+ */
+#    if defined(AARCH64) || defined(ARM)
+    signed char signals_pending; /* != 0: pending; < 0: currently handling one */
```

Comment:
I did some quick searches and couldn't find any other obviously incorrect usage of the default `char`. Mostly I saw buffers for filenames, commands or disassembled instrs

Some already use the explicit `signed char` where it makes sense, like:
https://github.com/DynamoRIO/dynamorio/blob/d7321a68934eb94e9746292ea4591dbf3a24e29f/core/arch/aarchxx/mangle.c#L660
https://github.com/DynamoRIO/dynamorio/blob/d7321a68934eb94e9746292ea4591dbf3a24e29f/core/ir/x86/decode_fast.c#L122

`globals_shared` typedefs a `signed char` as `sbyte` here: https://github.com/DynamoRIO/dynamorio/blob/d7321a68934eb94e9746292ea4591dbf3a24e29f/core/lib/globals_shared.h#L219
I can probably use `sbyte` instead of signed char in this PR too.

Reply:
Sure, it makes it clear it's unrelated to characters or strings.

Reply:
Also found #1034.